### PR TITLE
use zld for apple linker

### PR DESCRIPTION
**Pull request by NicolaLS**
this is suggested in the book "Zero to Production in Rust", and mold is currently broken

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -22,3 +17,11 @@ ar = "./.cargo/ar.x86_64-linux-android"
 [target.i686-linux-android]
 linker = "./.cargo/ld.i686-linux-android"
 ar = "./.cargo/ar.i686-linux-android"
+
+[profile.dev]
```
Comment (dpc) [inline .cargo/config.toml:21]: How does this work. Is this a section like any other, and the next ones apply to both debug and release builds? That's what it seems to me.

Comment (NicolaLS) [reply .cargo/config.toml:21]: I'm not sure tbh, naively put it in like this. I think it should behave like this. https://doc.rust-lang.org/cargo/reference/profiles.html#profile-selection

Comment (elsirion) [reply .cargo/config.toml:21]: Looking at the TOML spec that doesn't seem to be the way it works https://toml.io/en/v1.0.0#table

Comment (elsirion) [reply .cargo/config.toml:21]: More importantly, the semantics you do want to achieve don't seem to work:

```toml
[profile.dev.target.aarch64-apple-darwin]
rustflags = ["-C", "link-arg=-fuse-ld=zld"]
```

Comment (dpc) [reply .cargo/config.toml:21]: I looked briefly how to get it to work, and couldn't find any way.

Comment (elsirion) [reply .cargo/config.toml:21]: See https://github.com/rust-lang/cargo/issues/4897

**Comment by elsirion** [comment]
Does it make sense to turn on `zld` for apple in general and not only dev?

**Comment by NicolaLS** [comment]
> Does it make sense to turn on `zld` for apple in general and not only dev?

In the [repos](https://github.com/michaeleisel/zld) README it is recommended to only use it for debug builds. However reading everything I think we could go for it, after all every build from source right now is a "debug build", we could also add to our README that if somebody wants to actually use it they d need to change the `.cargo/config`.. I don't know if it's worth it though feel free to close this

**Comment by dpc** [comment]
We could do the opposite and check-in commented-out `zld` support for mac devs that want it.

What's the speedup anyway?

**Comment by NicolaLS** [comment]
> @NicolaLS We could potentially do the opposite and check-in commented-out `zld` support for mac devs that want it.
> 
> What's the speedup anyway?

lol yeah that sounds better. In the Readme they say "open an issue if its not 40% faster" (you need to do it twice the first time bc of caching)

**Review by dpc — APPROVED** [review]

**Pull request merged**